providers/datadog.rb

```diff
@@ -17,15 +17,26 @@ def emit(metrics)
     def send
       datadog_client.batch_metrics do
         pending.each do |metric, value|
-          metric = "#{metric_prefix}#{metric}"
-          puts "#{metric}: #{value}"
-          datadog_client.emit_point(metric, value, type: 'gauge')
+          metric_name, tags = extract_tags(metric)
+          metric_name = "#{metric_prefix}#{metric_name}"
+          puts "#{metric_name}#{tags ? "[#{tags.join(',')}]" : ''}: #{value}"
```

Comment:
Note: we're rejoining here rather than reusing the original `metric` so that logs clearly reflect exactly what was sent after the parsing.